R/remote.R

```diff
@@ -37,6 +37,54 @@ remote_name <- function(x) {
   lq$x$x
 }
 
+query_name <- function(x) {
+  UseMethod("query_name")
+}
+
+#' @export
+query_name.tbl_lazy <- function(x) {
+  query_name(x$lazy_query)
+}
+
+#' @export
+query_name.lazy_base_remote_query <- function(x) {
+  x$x
+}
+
+#' @export
+query_name.lazy_base_local_query <- function(x) {
+  ident(x$name)
+}
+
+#' @export
+query_name.lazy_query <- function(x) {
+  NULL
+}
+
+#' @export
+query_name.lazy_select_query <- function(x) {
+  if (!is_empty(x$where)) {
+    return()
+  }
+
+  if (!is_empty(x$order_by)) {
```

Comment:
Do you mind briefly explaining the reasoning in a comment? This is detecting if it will need to become a subquery?

Reply:
Actually, doesn't this function always return `NULL`?

Reply:
I can't recall why I added this method 😕 Maybe it also got obsolete after inlining `select()`. So, I just removed it 😄